### PR TITLE
Expose and use the proposed open_karto parameter, minimum time interval

### PR DESCRIPTION
**Pull request by spmaniato**
This PR is tied to ros-perception/open_karto#11 ⚠️ 

This is a follow-up to the open_karto PR mentioned above. It adds a new ROS param, `minimum_time_interval`, which is used to set the proposed open_karto param `MinimumTimeInterval`. It also passes the laser scan message's time stamp to open_karto enabling it to calculate the time interval between two scans.

### CODE REVIEW

File: src/slam_karto.cpp [hunk 1]
```diff
@@ -164,11 +164,15 @@ SlamKarto::SlamKarto() :
...
     mapper_->setParamUseScanMatching(use_scan_matching);
-  
+
   bool use_scan_barycenter;
   if(private_nh_.getParam("use_scan_barycenter", use_scan_barycenter))
     mapper_->setParamUseScanBarycenter(use_scan_barycenter);
 
+  double minimum_time_interval;
+  if(private_nh_.getParam("minimum_time_interval", minimum_time_interval))
+    mapper_->setParamMinimumTimeInterval(minimum_time_interval);
```
Comment (lucbettaieb) [inline src/slam_karto.cpp:174]: if you would, @spmaniato , please add `{}`s to the if/else's in this file and do a full lint.

**Review by lucbettaieb — CHANGES_REQUESTED** [review]
Seems legit.  I'd like to see better style in this file, so if you could, would you lint and add `{}`s?

Thanks!

**Comment by lucbettaieb** [comment]
@spmaniato Would you like to check this out?

**Comment by spmaniato** [comment]
@lucbettaieb Hey Luc, thanks for following up. I haven't worked on the Karto codebases in over a year. And unfortunately I don't have dev and sim environments set up to make and test any changes. And I don't feel comfortable pushing untested changes 😬 

Moreover, if we were to do a full lint of this codebase, I bet it would come up with hundreds of issues. Fixing just this one file wouldn't help much. And we would be mixing my minimal _functional_ changes with dozens of lines of linting-related changes, thus making the PR harder to review. May I recommend a separate PR that focuses entirely on linting and non-functional changes?

Sorry I cannot be more helpful at this time.

**Comment by lucbettaieb** [comment]
@spmaniato I didn't intend to ask you to do a full lint of the codebase.. lol that would be ridiculous.
If you want to get this PR in, let me know when you can test it, etc, and remove all of the extra spaces and stuff that your commits introduce into the code.  The changes are pretty minimal, so it shouldn't be much effort.

**Comment by spmaniato** [comment]
@lucbettaieb 

1. As I said, I do not have the dev or sim setup to re-test this PR, I'm sorry 😕 I had tested it back when I first opened it. See the corresponding (merged) PR description in `open_karto` for the results: https://github.com/ros-perception/open_karto/pull/11
2. My PR doesn't _introduce_  extra spaces. It _removes_ whitespace at the end of lines, which is pretty a standard thing that most editors do automatically these days. I don't see how that's harmful.

If you do not want to merge it in its current state, that's understandable. But I cannot work on it now or in the foreseeable future. Feel free to close it if you prefer and I'll re-open when and if I can work on it.

**Comment by lucbettaieb** [comment]
Oops - you're right about the spaces.  My bad!  Haven't looked at this in a while. :sweat_smile: 

Whenever you have time to test this, please let me know.  No rush.  Just wanted to ping you to see if you wanted to do anything with this.

**Comment by lucbettaieb** [comment]
@spmaniato Are you still interested in merging this?

**Pull request closed without merging**